### PR TITLE
Change: Use list / loop behavior for grammar exclusions. Minor exclusion updates / additions.

### PR DESCRIPTION
**Pull request by cfi-gb**
**What**:
- See title
- Parity PR to greenbone/vulnerability-tests#1201

**Why**:
Less lines and repetitive code

**How**:
N/A

**Checklist**:

- [ ] Tests
- [x] Conventional commit message
- [ ] Documentation

### CODE REVIEW

File: troubadix/plugins/grammar.py [hunk 1]
```diff
@@ -107,89 +107,61 @@ def check_for_false_positives(match: AnyStr, nasl_file: str) -> bool:
...
+        for known_fp in known_fps:
+
+            if isinstance(known_fp, re.Pattern):
+                return bool(known_fp.search(match))
+
+            elif isinstance(known_fp, str):
+                return known_fp in match
+
+            elif isinstance(known_fp, tuple):
+                (filename, content_fp) = known_fp
```
Comment (n-thumann) [inline troubadix/plugins/grammar.py:158]:
```suggestion
                filename, content_fp = known_fp
```

File: troubadix/plugins/grammar.py [hunk 2]
```diff
@@ -107,89 +107,61 @@ def check_for_false_positives(match: AnyStr, nasl_file: str) -> bool:
...
+                return filename in nasl_file and content_fp in match
+
+            else:
+                raise NotImplementedError(
+                    "Invalid type for known_fps entry. "
+                    "Valid types: str, re.Pattern, Tuple[str,str]"
```
Comment (n-thumann) [inline troubadix/plugins/grammar.py:164]:
```suggestion
                    "Valid types: str, re.Pattern, Tuple[str, str]"
```

**Review by n-thumann — COMMENTED** [review]
Minor styling preferences

**Comment by cfi-gb** [comment]
> @cfi-gb I did another rework of this that I think is even more future proof, I would be open to discussion / rejection about this though.

Ah, great. Thanks a lot for taking care :+1: 

Unfortunately i'm not really able to judge / review these changes so i would like to hand over to @n-thumann for final review and approval.

**Comment by mbrinkhoff** [comment]
> > @cfi-gb I did another rework of this that I think is even more future proof, I would be open to discussion / rejection about this though.
> 
> Ah, great. Thanks a lot for taking care +1
> 
> Unfortunately i'm not really able to judge / review these changes so i would like to hand over to @n-thumann for final review and approval.
> 
> I guess we would also need to update the PR title on the changed functionality?

Done, since Nicolas already reviewed the changes, I will approve now.

**Pull request merged**